doc/namespace-store-crd.md

```diff
@@ -0,0 +1,218 @@
+
+
+# Reconcile
+
+#### AWS-S3 type
+
+Create a cloud resource within the NooBaa brain and use S3 API for reading or writing data in the AWS cloud.
+```shell
+noobaa -n noobaa namespacestore create aws-s3 bs --access-key KEY --secret-key SECRET --target-bucket BUCKET
+```
+```yaml
+apiVersion: noobaa.io/v1alpha1
+kind: NamespaceStore
+metadata:
+  finalizers:
+  - noobaa.io/finalizer
+  labels:
+    app: noobaa
+  name: bs
+  namespace: noobaa
+spec:
+  awsS3:
+    secret:
+      name: namespace-store-aws-s3-bs
+      namespace: noobaa
+    targetBucket: BUCKET
+  type: aws-s3
+```
+
+#### S3-COMPATIBLE type
+
+Create a cloud resource within the NooBaa brain and use S3 API for reading or writing data in any S3 API compatible endpoint.
+```shell
+noobaa -n noobaa namespacestore create s3-compatible bs --endpoint ENDPOINT --signature-version v4 --access-key KEY --secret-key SECRET --target-bucket BUCKET
+```
+```yaml
+apiVersion: noobaa.io/v1alpha1
+kind: NamespaceStore
+metadata:
+  finalizers:
+  - noobaa.io/finalizer
+  labels:
+    app: noobaa
+  name: bs
+  namespace: noobaa
+spec:
+  s3Compatible:
+    endpoint: ENDPOINT
+    secret:
+      name: namespace-store-s3-compatible-bs
+      namespace: noobaa
+    signatureVersion: v4
+    targetBucket: BUCKET
+  type: s3-compatible
+```
+
+#### IBM-COS type
+
+Create a cloud resource within the NooBaa brain and use IBM COS API for reading or writing data in any IBM COS API compatible endpoint.
+```shell
+noobaa -n noobaa namespacestore create ibm-cos bs --endpoint ENDPOINT --access-key KEY --secret-key SECRET --target-bucket BUCKET
+```
+```yaml
+apiVersion: noobaa.io/v1alpha1
+kind: NamespaceStore
+metadata:
+  finalizers:
+  - noobaa.io/finalizer
+  labels:
+    app: noobaa
+  name: bs
+  namespace: noobaa
+spec:
+  s3Compatible:
+    endpoint: ENDPOINT
+    secret:
+      name: namespace-store-ibm-cos-bs
+      namespace: noobaa
+    signatureVersion: v2
+    targetBucket: BUCKET
+  type: ibm-cos
+```
+
+#### AZURE-BLOB type
+
+Create a cloud resource within the NooBaa brain and use BLOB API for reading or writing data in Azure cloud.
+```shell
+noobaa -n noobaa namespacestore create azure-blob bs --account-key KEY --account-name NAME --target-blob-container CONTAINER
+```
+```yaml
+apiVersion: noobaa.io/v1alpha1
+kind: NamespaceStore
+metadata:
+  finalizers:
+  - noobaa.io/finalizer
+  labels:
+    app: noobaa
+  name: bs
+  namespace: noobaa
+spec:
+  azureBlob:
+    secret:
+      name: namespace-store-azure-blob-bs
+      namespace: noobaa
+    targetBlobContainer: CONTAINER
+  type: azure-blob
+```
+
+#### Credentials change
+
+In case the credentials of a Namespace-store need to be updated due to a periodic security policy or concern, the appropriate secret should be updated by the user, and the operator will be responsible for watching changes in those secrets and propagating the new credential update to the NooBaa system server.
+
```

Comment:
```suggestion
In case the credentials of a Namespace-store need to be updated due to periodic security policy or concern, the appropriate secret should be updated by the user, and the operator will be responsible for watching changes in those secrets and propagating the new credential update to the NooBaa system server.

```